va/http.go

```diff
@@ -648,6 +648,14 @@ func (va *ValidationAuthorityImpl) processHTTPValidation(
 		return nil, records, newIPError(target, berrors.UnauthorizedError("Invalid response from %s: %q",
 			records[len(records)-1].URL, body))
 	}
+
+	// We were successful, so record the negotiated key exchange mechanism in the
+	// last validationRecord.
+	// TODO(#7321): Remove this when we have collected enough data.
+	if httpResponse.TLS != nil {
+		records[len(records)-1].UsedRSAKEX = usedRSAKEX(httpResponse.TLS.CipherSuite)
+	}
```

Comment:
Teeechnically this should happen in processRedirect too, since we could get an HTTP->HTTPS->HTTPS or HTTP->HTTPS->HTTP redirect where the middle hope used RSA KEX. But that's probably very rare indeed so I'm okay skipping it.

Reply:
Yeah, and it's pretty unclear to me _where_ in processRedirect would be the right place to put it, esp if we only care about this value in the case that validation succeeds.

Reply:
It should happen right around line 511:

		if req.Response.TLS != nil && req.Response.TLS.Version < tls.VersionTLS12 {

And it's okay to set this field even for a validation that might not succeed; we have a summary loop at the end anyhow.